lib/CMakeLists.txt

```diff
@@ -26,37 +27,18 @@ set(OPTSCHED_SRCS Scheduler/aco.cpp
   Wrapper/OptSchedMachineWrapper.cpp
   Wrapper/OptSchedDDGWrapperBasic.cpp
   Wrapper/OptSchedGenericTarget.cpp
-)
+  Wrapper/AMDGPU/GCNOptSched.cpp
+  Wrapper/AMDGPU/OptSchedGCNTarget.cpp
+  Wrapper/AMDGPU/OptSchedDDGWrapperGCN.cpp
```

Comment:
What about CPU builds that don't support the AMDGPU?

Reply:
I haven't put too much thought into CPU builds -- but I agree this is not very extensible -- probably use a variable to hold sources that are dependent on what we are building for.

Reply:
Something you can try: [a generator expression](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html) against `OPTSCHED_ENABLE_AMDGPU`. I think that would be something like this:

```cmake
set(AMDGPU_SOURCES
  Wrapper/AMDGPU/GCNOptSched.cpp
  Wrapper/AMDGPU/OptSchedGCNTarget.cpp
  Wrapper/AMDGPU/OptSchedDDGWrapperGCN.cpp
)

add_llvm_target(
  OptSched
  Scheduler/aco.cpp
  # ...
  Wrapper/OptSchedGenericTarget.cpp
  $<$<BOOL:${OPTSCHED_ENABLE_AMDGPU}>:${AMDGPU_SOURCES}>
)
```

Reply:
At the present, it seems like we are fine with an ordinary CMake variable. If / when we want to support a more flexible build system (e.g. targeting x86 vs AMDGPU backends on the fly) I can see how having more customized generations would be extremely helpful. 

For this PR though, I think I'll just stick with the simple variable.